File: jan13/dec2bin.c
#include <stdio.h>

void dec2bin(int n) {

    while (n > 0) {
	printf("%d", n % 2);
        n = n / 2;
    }

    printf("\n");
}

int main() {

    dec2bin(333); // 101001101

    return 0;
}

File: float/dec2bin.py

def dec2bin(f: float) -> str:

    r = ""

    for i in range(64):
        f = 2*f
        if f >= 1.0:
            r = r + '1'
            f = f - 1
        else:
            r = r + '0'

    return r

if __name__ == "__main__":
    print(dec2bin(.14159))
